src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorMvcOptionsSetup.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    public class ApiBehaviorMvcOptionsSetup : IConfigureOptions<MvcOptions>
+    {
+        private readonly IErrorDescriptionFactory _errorDescriptionFactory;
+
+        public ApiBehaviorMvcOptionsSetup(IErrorDescriptionFactory errorDescriptionFactory)
+        {
+            _errorDescriptionFactory = errorDescriptionFactory;
+        }
+
+        public void Configure(MvcOptions options)
+        {
+            options.ApiBehavior.InvalidModelStateResponseFactory = GetInvalidModelStateResponse;
+
+            IActionResult GetInvalidModelStateResponse(ActionContext context)
+            {
+                var errorDetails = _errorDescriptionFactory.CreateErrorDescription(
+                    context.ActionDescriptor, 
+                    new ValidationProblemDetails(context.ModelState));
```

Comment:
this needs to call in to your new service

Reply:
ErrorDescriptionFactory?